### ISSUE TITLE
Intermittent fail in CI - test_snapshot_basic.py::test_5_full_snapshots

### ISSUE THREAD

**Issue by dianpopa** [firecracker-microvm/firecracker#2075]
CI failed on a PR that does not impact failing test. See https://firecracker-pr.s3.amazonaws.com/pr-logs/x86_64/02074/dianpopa-firecracker-4413f28.log.html

**Comment by dianpopa** [comment]
Another unrelated fail was noticed here: https://firecracker-pr.s3.amazonaws.com/pr-logs/x86_64/02070/dianpopa-firecracker-bcfbb2a.log.html.

**Comment by acatangiu** [comment]
The `fio` part of the test is unreliable with sporadic fails:

```bash
____________________________ test_5_full_snapshots _____________________________
integration_tests/functional/test_snapshot_basic.py:278: in test_5_full_snapshots
    test_matrix.run_test(_test_seq_snapshots)
framework/matrix.py:151: in run_test
    self._backtrack(test_fn, [])
framework/matrix.py:126: in _backtrack
    self._backtrack(test_fn, cartesian_product)
framework/matrix.py:126: in _backtrack
    self._backtrack(test_fn, cartesian_product)
framework/matrix.py:126: in _backtrack
    self._backtrack(test_fn, cartesian_product)
framework/matrix.py:115: in _backtrack
    self._run_test_fn(cartesian_product, test_fn)
framework/matrix.py:137: in _run_test_fn
    test_fn(self._context)
integration_tests/functional/test_snapshot_basic.py:180: in _test_seq_snapshots
    _guest_run_fio_iteration(ssh_connection, i)
integration_tests/functional/test_snapshot_basic.py:24: in _guest_run_fio_iteration
    assert exit_code == 0
E   assert 255 == 0
```

**Comment by sandreim** [comment]
Fixed by PR #2067

**Comment by iulianbarbu** [comment]
This occurred again for #2226 .

**Comment by alindima** [comment]
@iulianbarbu it would be better to post the link to the CI output. the CI status check on the PR will change in the future.

**Comment by iulianbarbu** [comment]
Here is the CI output: https://buildkite.com/firecracker/firecracker-ci-amd/builds/1304#1cd747da-63c7-45bc-b50f-9dbab79edd8f.

**Comment by dianpopa** [comment]
This fail has not been seen in a while. It should have been fixed by #2503.
